gems/smithy-schema/spec/smithy-schema/shapes_spec.rb

```diff
@@ -29,6 +29,64 @@ module Shapes
         end
       end
 
+      describe Structure do
+        subject { Structure.new }
+
+        it 'is a subclass of Shape' do
+          expect(subject).to be_kind_of(Shape)
+        end
+
+        describe '#initialize' do
+          it 'defaults members to empty hash' do
+            expect(subject.members).to be_empty
+          end
+
+          it 'defaults names_by_member_name to empty hash' do
+            expect(subject.names_by_member_name).to be_empty
+          end
+        end
+
+        describe '#add_member' do
+          it 'adds a member' do
+            subject.add_member(:foo, 'foo', StringShape.new)
+            expect(subject.members[:foo]).to be_kind_of(MemberShape)
+          end
+
+          it 'can set traits on the member' do
+            subject.add_member(:foo, 'foo', StringShape.new, traits: { 'trait' => 'value' })
+            expect(subject.members[:foo].traits).to eq({ 'trait' => 'value' })
+          end
+        end
+
+        describe '#member?' do
+          it 'returns true if member exists' do
+            subject.add_member(:foo, 'foo', StringShape.new)
+            expect(subject.member?(:foo)).to be(true)
+          end
+        end
+
+        describe '#member' do
+          it 'returns the member' do
+            subject.add_member(:foo, 'foo', StringShape.new)
+            expect(subject.member(:foo)).to be_kind_of(MemberShape)
+          end
+        end
+
+        describe '#name_by_member_name?' do
+          it 'returns true if member by member name exists' do
+            subject.add_member(:foo, 'foo', StringShape.new)
```

Comment:
Could update the member name arg to be like `fOO` to ensure that we can reference it back.